Checkout: Display tax breakdown when available

danielmx-dev opened the pull request:
#### Changes proposed in this Pull Request

* Display the breakdown of the individual applicable taxes when available in the shopping cart response.

![image](https://user-images.githubusercontent.com/15204776/120828318-01853880-c522-11eb-9af5-0b77f8bc697e.png)


#### Testing instructions
 * Apply D62284-code
 * While Sandboxed, add any product to the cart and use the following billing information: Country: Canada, Postal Code: G0T 9Z9
 * Verify that the taxes are rendered properly, as in the image above.

danielmx-dev committed:
Checkout: Display tax breakdown when available

sirbrillig committed:
Add total_tax_breakdown to getEmptyResponseCart

danielmx-dev commented on packages/wpcom-checkout/src/transformations.ts:
```diff
@@ -87,6 +87,37 @@ export function getTaxLineItemFromCart( responseCart: ResponseCart ): LineItem |
+		( taxBreakdownItem: TaxBreakdownItem, index: number ): LineItem => {
+			const id = `tax-line-item-${ index }`;
+			const label = taxBreakdownItem.label
+				? `${ taxBreakdownItem.label } (${ taxBreakdownItem.rate_display })`
```
I need some feedback about whether this label needs to be localized.

sirbrillig replied:
These seem like things that do not need to be localized here; I think that the `label` should be localized on the backend (if they even need localization) and `rate_display` is mostly numeric so it should be fine.

sirbrillig reviewed (approved):
Looks good to me. The instructions work as described (although the tax items I got were different, but I imagine that's expected):

<img width="930" alt="Screen Shot 2021-06-04 at 4 24 27 PM" src="https://user-images.githubusercontent.com/2036909/120859008-994e4b00-c551-11eb-852b-b7a3210cbfed.png">

The total tax adds up to the tax items. I'm assuming you only want to display the individual taxes in the sidebar, and not at the end? Worth noting that at mobile width the sidebar is collapsed by default, so this information will be hidden unless the user clicks to expand it.

<img width="561" alt="Screen Shot 2021-06-04 at 4 28 53 PM" src="https://user-images.githubusercontent.com/2036909/120859286-ff3ad280-c551-11eb-850f-121e54d4ca45.png">

<img width="566" alt="Screen Shot 2021-06-04 at 4 29 05 PM" src="https://user-images.githubusercontent.com/2036909/120859297-0235c300-c552-11eb-8cad-eebd43906a4e.png">

danielmx-dev commented:
>  I'm assuming you only want to display the individual taxes in the sidebar, and not at the end? 

Now that you mention it I'm not sure if it'd be better to also display the breakdown in every step as well for consistency.

This is how it would look like 
![Screen Capture on 2021-06-04 at 16-10-31](https://user-images.githubusercontent.com/15204776/120863166-b8e47400-c54f-11eb-846c-18791a5a564c.gif)

What do you think @sirbrillig  ? The maximum number of tax lines we expect to display is 2, so I guess it still looks fine.

sirbrillig commented:
Nice! I think that looks better for the last step to include the tax breakdown... but this PR has made me wonder why we display the taxes at all on the (edited) first step. It seems out-of-place and may just be leftover from when we originally had that step at the end of checkout. IMO we should just get rid of the taxes line there and leave the other two. What do you think about that? (I'm fine with it either way)

danielmx-dev committed:
Add tax breakdown to payment method step

danielmx-dev committed:
Remove tax line items from "Your order" section

danielmx-dev committed:
Merge branch 'update/show-tax-breakdown-in-checkout' of github.com:Automattic/wp-calypso into update/show-tax-breakdown-in-checkout

danielmx-dev commented:
> Nice! I think that looks better for the last step to include the tax breakdown... but this PR has made me wonder why we display the taxes at all on the (edited) first step. It seems out-of-place and may just be leftover from when we originally had that step at the end of checkout. IMO we should just get rid of the taxes line there and leave the other two. What do you think about that? (I'm fine with it either way)

I removed the tax line items from the order review section and added the breakdown to the payment method step as suggested. I'll appreciate if you could take another look when you have a chance. 🙏

sirbrillig reviewed (approved):
Looks great. Thank you!
